components/script/dom/clipboard.rs

```diff
@@ -40,6 +90,89 @@ impl Clipboard {
 }
 
 impl ClipboardMethods<crate::DomTypeHolder> for Clipboard {
+    /// <https://w3c.github.io/clipboard-apis/#dom-clipboard-readtext>
+    fn ReadText(&self, can_gc: CanGc) -> Rc<Promise> {
+        // Step 1 Let realm be this's relevant realm.
+        // Step 2 Let p be a new promise in realm.
+        let p = Promise::new(&self.global(), can_gc);
+
+        // Step 3 Run the following steps in parallel:
```

Comment:
What you;d have to do here, is send the message to the embedder, and then attach a handler to the IPC route to then queue the task and continue on the script thread. 

You could use https://github.com/servo/servo/blob/b1d0b6a37bed9512805cfab07da15cca8f750238/components/script/routed_promise.rs#L17

This will require changing the embedder msg to include a sender. See other uses of `routed_promise` for examples.